tests/lean/run/8894.lean

```diff
@@ -0,0 +1,50 @@
+open Lean.Order
+
+def A : Type := Prop
+def B : Type := Prop
+
+instance : Lean.Order.PartialOrder A where
+  rel := sorry
+  rel_refl := sorry
+  rel_trans := sorry
+  rel_antisymm := sorry
+
+instance : Lean.Order.PartialOrder B where
+  rel := sorry
+  rel_refl := sorry
+  rel_trans := sorry
+  rel_antisymm := sorry
+
+instance : Lean.Order.CCPO A where
+  csup := sorry
+  csup_spec := sorry
+
+instance : Lean.Order.CCPO B where
+  csup := sorry
+  csup_spec := sorry
+
+/--
+error: Could not prove 'tick' to be monotone in its recursive calls:
+  Cannot eliminate recursive call in
+    tock (n + 1)
```

Comment:
This is not a very helpful error message, and we do have a bit more context. So one could throw a different error message here and explain that this is likely due to different orders. 

But since this is really an obscure corner and the tracing option exists, good for now.